libbess-python/bess.py

```diff
@@ -210,9 +211,12 @@ def check_constraints(self):
             for module in response.modules:
                 print 'constraints violated for module %s --'\
                     ' please check bessd log' % module.name
+            error = True
         if response.fatal:
             raise self.ConstraintError("Fatal violation of "
                                        "scheduling constraints")
+            error = True
```

Comment:
These lines will never be executed since the control flow already left this function. If you want a tri-state return value, how about something like:
* success: no return value
* soft/hard failure: raise ConstraintError("Scheduling violation(s) found", violations=response.violations, modules=response.modules, fatal=response.fatal)

So that the caller (e.g., bessctl) can deal with the error programmatically? The caller may want to make sure that there are even no minor errors, choose its own error message mechanism (e.g., `cli.err()`), etc.